packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,459 @@
+}>(({ odysseyDesignTokens }) => ({
+  display: "flex",
+  alignItems: "center",
+  marginRight: odysseyDesignTokens.Spacing3,
+  borderRight: `${odysseyDesignTokens.BorderWidthMain} solid ${odysseyDesignTokens.HueNeutral200}`,
+}));
+
+const LinkContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  paddingRight: odysseyDesignTokens.Spacing3,
+  "& a": {
+    color: `${odysseyDesignTokens.TypographyColorHeading} !important`,
+  },
+}));
+
+const TopNavContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  display: "flex",
+  alignItems: "center",
+  backgroundColor: odysseyDesignTokens.HueNeutralWhite,
+  height: odysseyDesignTokens.Spacing9,
+}));
+
+const LogoContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  display: "flex",
+  alignItems: "center",
+  padding: `0 ${odysseyDesignTokens.Spacing9} 0 ${odysseyDesignTokens.Spacing5}`,
+}));
+
+const SearchFieldContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  width: "350px",
+  padding: `${odysseyDesignTokens.Spacing2} ${odysseyDesignTokens.Spacing3}`,
+}));
+
+const AdditionalNavItemContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  padding: `0 ${odysseyDesignTokens.Spacing3}`,
+}));
+
+const TopNav = ({
+  hasLogo,
+  SearchFieldComponent,
+  topNavLinkItems,
+  AdditionalNavItemComponent,
+  settingsLink,
+  helpLink,
+  userProfile,
+}: TopNavProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const LogoStyles = useMemo(
+    () => ({
+      fontSize: odysseyDesignTokens.TypographyScale6,
+    }),
+    [odysseyDesignTokens],
+  );
+
+  const LogoWordmarkStyles = useMemo(
+    () => ({
+      width: "55px",
+      height: "20px",
+      paddingLeft: odysseyDesignTokens.Spacing2,
+    }),
+    [odysseyDesignTokens],
+  );
+
+  return (
+    <TopNavContainer odysseyDesignTokens={odysseyDesignTokens}>
+      {hasLogo && (
+        <LogoContainer odysseyDesignTokens={odysseyDesignTokens}>
+          <AuraIcon sx={LogoStyles} />
+          <AuraWordmarkIcon sx={LogoWordmarkStyles} />
+        </LogoContainer>
+      )}
+      {SearchFieldComponent && (
+        <SearchFieldContainer odysseyDesignTokens={odysseyDesignTokens}>
+          {SearchFieldComponent}
+        </SearchFieldContainer>
+      )}
+      <TopNavListContainer>
+        {topNavLinkItems?.map((item) => {
+          return <TopNavItemContent key={item.id} {...item} />;
+        })}
```

Comment:
It's easier to read these if we use implicit return as that matches the rest of the codebase:
```suggestion
        {topNavLinkItems?.map((item) => (
          <TopNavItemContent {...item} key={item.id} />;
        ))}
```

Also, to keep things consistent, we put the `{...props}` part before any manually-defined props. That will give you the correct code you're expecting. If `item` had a `key` prop, it would override the `{...item}` code.